src/date-picker/index.ts

```diff
@@ -84,6 +83,11 @@ let exportDatePicker;
       return true;
     }
 
+    protected willUpdate(changedProperties: PropertyValues) {
+      this._updateErrorText();
```

Comment:
There is a case:
1. Input an invalid value to show "Format is not valid." error on UI.
2. Set the valid value in JS code.
=> "Format is not valid." error is still displayed.

Cause:
The value of "_errorText" is set in willUpdate(), but the value of "_errorFormat" is reset in update().
update() is called after willUpdate(), so the new value of "_errorFormat" is not used.

Suggestion:
Move the content of update() to willUpdate().
Call "this._updateErrorText();" after setting the value of "_errorFormat".